### PR TITLE
[9.0] Port sale_start_end_dates and sale_rental

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

File: sale_rental/data/rental_data.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  © 2014-2016 Akretion (http://www.akretion.com/)
+  @author Alexis de Lattre <alexis.delattre@akretion.com>
+  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
+
+<odoo noupdate="0">
```
Comment (atchuthan) [inline sale_rental/data/rental_data.xml]: Could remove noupdate="0" attribute?

**Review by atchuthan — APPROVED** [review]
Minor changes
Runbot :+1:

**Review by dvdhinesh — APPROVED** [review]
Looks good except the @api.one deprecated warning on lint check. 
Code + Functional review :+1:

**Comment by alexis-via** [comment]
@api.one is not deprecated any more ; this needs to be fixed in the pylint config file.

**Review by gurneyalex — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@alexis-via, I'm afraid that the final decision has been to deprecate @api.one.

**Comment by alexis-via** [comment]
@pedrobaeza It changes too often for me to follow... Maybe in a few weeks/months they will change their mind again ? :)

**Comment by pedrobaeza** [comment]
I don't think so this time. It was discussed in the Odoo experience and they are pretty sure to go on with the deprecation.

**Comment by lasley** [comment]
Here's the ticket where we finally got the confirmation as well - https://github.com/odoo/odoo/pull/13199

**Review by lasley — APPROVED** [review]
LGTM outside of the api.one usage that has already been noted. Code, no test

**Comment by pedrobaeza** [comment]
Merging as there's no special problem in use @api.one (it's also available in v10)

**Comment by dvdhinesh** [comment]
@alexis-via Thanks for the PR. Any idea of porting this to V10?

**Comment by alexis-via** [comment]
@dvdhinesh It is certainly a good idea ! Feel free to propose a PR if you have the time to do it.

**Comment by dvdhinesh** [comment]
@alexis-via sure.

**Comment by dvdhinesh** [comment]
@alexis-via Can you please review this PR #357

**Pull request merged**